src/test/java/com/atlassian/bitbucket/jenkins/internal/status/LocalSCMListenerTest.java

```diff
@@ -62,25 +75,27 @@ public void setup() throws URISyntaxException {
         when(bitbucketSCM.getBitbucketSCMRepository()).thenReturn(scmRepository);
         when(repositoryHelper.getRepository(any(), eq(bitbucketSCM))).thenReturn(scmRepository);
         when(repositoryHelper.getRepository(any(), eq(gitSCM))).thenReturn(scmRepository);
-        listener = spy(new LocalSCMListener(buildStatusPoster, repositoryHelper));
+        listener = spy(new LocalSCMListener(buildStatusPoster, librariesProvider, repositoryHelper));
+        when(librariesProvider.get()).thenReturn(globalLibraries);
     }
 
     @Test
-    public void testOnCheckoutWithNonGitSCMDoesNotPostBuildStatus() {
-        SCM scm = mock(SCM.class);
-        when(repositoryHelper.getRepository(run, scm)).thenReturn(scmRepository);
-
-        listener.onCheckout(run, scm, null, taskListener, null, null);
-
-        verify(buildStatusPoster, never()).postBuildStatus(any(), any(), any());
-    }
-
-    @Test
-    public void testOnCheckoutWithNoRepositoryDoesNotPostBuildStatus() {
-        SCM scm = mock(SCM.class);
-
-        listener.onCheckout(run, scm, null, taskListener, null, null);
-
+    public void testOnCheckoutMultiBranchPipelineWithFolderLibraryDoesNotPostBuildStatus() throws IOException {
+        when(bitbucketSCM.getId()).thenReturn("SomeId");
+        WorkflowMultiBranchProject workflowMultiBranchProject =
+                jenkinsRule.getInstance().createProject(WorkflowMultiBranchProject.class, "MultiBranchProject");
+        WorkflowJob workflowJob = new WorkflowJob(workflowMultiBranchProject, "Job1");
+        when(run.getParent()).thenReturn(workflowJob);
+        LibraryConfiguration libConfig = mock(LibraryConfiguration.class);
+        SCMRetriever scmRetriever = mock(SCMRetriever.class);
+        when(libConfig.getRetriever()).thenReturn(scmRetriever);
+        when(scmRetriever.getScm()).thenReturn(bitbucketSCM);
+        FolderLibraries folderLibraries = new FolderLibraries(singletonList(libConfig));
+        workflowMultiBranchProject.addProperty(folderLibraries);
+
+        listener.onCheckout(run, bitbucketSCM, null, taskListener, null, null);
+
+        verify(bitbucketSCM, times(2)).getId();
```

Comment:
Why is this exactly two? If this is just part of the flow of the test and you just want to verify some number, I think `atLeast(1)` might be better so it doesn't break tests in the future, but if that number has a specific meaning here then times(2) should be fine

Reply:
On line 165 I put a comment about why it's 2. I have added comments on all the other `times(2)`s to add some clarity.